web-ui/src/components/reviews/TeamReviews.jsx

```diff
@@ -621,6 +625,37 @@ const TeamReviews = ({ onBack, periodId }) => {
     }
   };
 
+  const requestApproval = async () => {
+    const msg = validateReviewPeriod(period);
+    setValidationMessage(msg);
+    if (msg) return;
+
+    const visibleIds = new Set(visibleTeamMembers().map(m => m.id));
```

Comment:
If there are no validation errors, we now display a modal dialog whose text depends on the number of unapproved review assignments.